server-session/src/main/java/com/iluwatar/sessionserver/App.java

```diff
@@ -75,37 +78,34 @@ public static void main(String[] args) throws IOException {
     server.start();
 
     // Start background task to check for expired sessions
-    sessionExpirationTask();
+    startSessionExpirationTask();
 
     LOGGER.info("Server started. Listening on port 8080...");
   }
 
-  private static void sessionExpirationTask() {
-    new Thread(() -> {
-      while (true) {
-        try {
-          LOGGER.info("Session expiration checker started...");
-          Thread.sleep(SESSION_EXPIRATION_TIME); // Sleep for expiration time
-          Instant currentTime = Instant.now();
-          synchronized (sessions) {
-            synchronized (sessionCreationTimes) {
-              Iterator<Map.Entry<String, Instant>> iterator =
-                  sessionCreationTimes.entrySet().iterator();
-              while (iterator.hasNext()) {
-                Map.Entry<String, Instant> entry = iterator.next();
-                if (entry.getValue().plusMillis(SESSION_EXPIRATION_TIME).isBefore(currentTime)) {
-                  sessions.remove(entry.getKey());
-                  iterator.remove();
-                }
+  private static void startSessionExpirationTask() {
+    ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+    Runnable sessionExpirationChecker = () -> {
+      try {
+        LOGGER.info("Session expiration checker started...");
+        Instant currentTime = Instant.now();
+        synchronized (sessions) {
+          synchronized (sessionCreationTimes) {
```

Comment:
Could we use ConcurrentHashMap for sessions and sessionCreationTimes allowing us to remove the need for synchronization blocks?